lib/zenaton/interfaces/task.rb

```diff
@@ -9,7 +10,35 @@ class Task < Job
       # Child classes should implement the handle method
       def handle
         raise NotImplemented,
-              "Your workflow does not implement the `handle' method"
+              "Your workflow does not implement the `#handle' method"
+      end
+
+      # (Optional) Implement this method for automatic retrial of task in
+      # case of failures.
+      # @param _exception [Exception] the reason for the task failure.
+      # @return [#negative?, FalseClass, NilClass] the non-negative amount of
```

Comment:
What does the `#negative?` means in that case?
I don't understand why you had to change it from positive to negative.

Reply:
`#negative?` means something that responds to the instance method with the same name. We've change the check we do on the agent from `#positive?` do `#negative?` to allow 0 as an allowed value. I've updated the docs to inform the user of the method we'll be using against the returned value from this.

Reply:
Yes but we also changed from an `if` to an `unless`.
When I read this, I understand that I can return a negative number, false, or nil. Which is not true. But maybe I misunderstand it because I'm not used to it.